kybra/compiler/kybra_post_install/src/main.rs

```diff
@@ -0,0 +1,402 @@
+    canister_name: &str,
+    max_chunk_size: usize,
+    dfx_network: &str,
+) -> Result<(), String> {
+    let python_stdlib_bytecode = get_python_stdlib_bytecode()?;
+
+    let local_python_stdlib_bytecode_hash =
+        get_local_python_stdlib_bytecode_hash(&python_stdlib_bytecode);
+
+    let remote_python_stdlib_bytecode_hash =
+        get_remote_python_stdlib_bytecode_hash(canister_name, dfx_network)?;
+
+    let hashes_do_not_match =
+        format!("(\"{local_python_stdlib_bytecode_hash}\")") != remote_python_stdlib_bytecode_hash;
+
+    if hashes_do_not_match {
+        handle_python_stdlib_hashes_do_not_match(
+            canister_name,
+            dfx_network,
+            python_stdlib_bytecode,
+            max_chunk_size,
+        )?;
+    }
+
+    Ok(())
+}
+
+fn get_local_python_stdlib_bytecode_hash(python_stdlib_bytecode: &Vec<u8>) -> String {
+    let mut hasher = Sha256::new();
+    hasher.update(python_stdlib_bytecode);
+    let result = hasher.finalize();
+    hex::encode(result)
+}
+
+fn get_remote_python_stdlib_bytecode_hash(
+    canister_name: &str,
+    dfx_network: &str,
+) -> Result<String, String> {
+    let remote_python_stdlib_hash_output = Command::new("dfx")
+        .arg("canister")
+        .arg("call")
+        .arg("--network")
+        .arg(dfx_network)
+        .arg(canister_name)
+        .arg("python_stdlib_hash")
+        .output()
+        .map_err(|_| create_error_string("Failed to execute the dfx canister call command"))?;
+
+    if !remote_python_stdlib_hash_output.status.success() {
+        return Err(create_error_string(&String::from_utf8_lossy(
+            &remote_python_stdlib_hash_output.stderr,
+        )));
+    }
+
+    let remote_python_stdlib_bytecode_hash =
+        String::from_utf8_lossy(&remote_python_stdlib_hash_output.stdout)
+            .trim()
+            .to_string();
+
+    Ok(remote_python_stdlib_bytecode_hash)
+}
+
+fn handle_python_stdlib_hashes_do_not_match(
+    canister_name: &str,
+    dfx_network: &str,
+    python_stdlib_bytecode: Vec<u8>,
+    max_chunk_size: usize,
+) -> Result<(), String> {
+    let python_stdlib_bytecode_chunks = split_into_chunks(python_stdlib_bytecode, max_chunk_size);
+
+    for (index, python_stdlib_bytecode_chunk) in python_stdlib_bytecode_chunks.iter().enumerate() {
+        upload_chunk(
+            "Python stdlib",
+            canister_name,
+            python_stdlib_bytecode_chunk,
+            "upload_python_stdlib_chunk",
+            dfx_network,
+            index,
+            python_stdlib_bytecode_chunks.len(),
+        )?;
+    }
+
+    Ok(())
+}
+
+fn get_python_stdlib_bytecode() -> Result<Vec<u8>, String> {
+    #[cfg(ic_build)]
+    let kybra_version = std::env::var("KYBRA_VERSION")
+        .map_err(|_| create_error_string("KYBRA_VERSION environment variable not present"))?;
+
+    #[cfg(ic_build)]
+    let python_stdlib_path = dirs::home_dir()
+        .ok_or(create_error_string("Home directory not found"))?
```

Comment:
I wonder if people will have issue with this in a ci/cd env. Like, I wonder if they'll by trying to build in production in some kubernetes pod that doesn't have a home dir set or something...